test/Grpc.IntegrationTests/OrchestrationPatterns.cs

```diff
@@ -69,562 +69,705 @@ public async Task ScheduleOrchesrationWithTags()
+                        throw new ArgumentNullException(nameof(input));
+                    }
+
+                    return ctx.CallActivityAsync<JsonNode>("SpecialSerialization_Activity", input);
+                })
+                .AddActivityFunc<JsonNode, JsonNode?>("SpecialSerialization_Activity", (ctx, input) =>
+                {
+                    if (input is not null)
+                    {
+                        input["newProperty"] = "new value";
+                    }
+
+                    return Task.FromResult(input);
+                }));
+        });
+
+        JsonNode input = new JsonObject() { ["originalProperty"] = "original value" };
+        string instanceId = await server.Client.ScheduleNewOrchestrationInstanceAsync(
+            "SpecialSerialization_Orchestration", input: input);
+        OrchestrationMetadata result = await server.Client.WaitForInstanceCompletionAsync(
+            instanceId, getInputsAndOutputs: true, this.TimeoutToken);
+        JsonNode? output = result.ReadOutputAs<JsonNode>();
+
+        Assert.NotNull(output);
+        Assert.Equal("original value", output?["originalProperty"]?.ToString());
+        Assert.Equal("new value", output?["newProperty"]?.ToString());
+    }
+
+    // TODO: Additional versioning tests
+    [Fact]
+    public async Task OrchestrationVersionPassedThroughContext()
+    {
+        var version = "0.1";
+        await using HostTestLifetime server = await this.StartWorkerAsync(b =>
+        {
+            b.AddTasks(tasks => tasks
+                .AddOrchestratorFunc<string, string>("Versioned_Orchestration", (ctx, input) =>
+                {
+                    return ctx.CallActivityAsync<string>("Versioned_Activity", ctx.Version);
+                })
+                .AddActivityFunc<string, string>("Versioned_Activity", (ctx, input) =>
+                {
+                    return $"Orchestration version: {input}";
+                }));
+        }, c =>
+        {
+            c.UseDefaultVersion(version);
+        });
+
+        var instanceId = await server.Client.ScheduleNewOrchestrationInstanceAsync("Versioned_Orchestration", input: string.Empty);
+        var result = await server.Client.WaitForInstanceCompletionAsync(instanceId, getInputsAndOutputs: true, this.TimeoutToken);
+        var output = result.ReadOutputAs<string>();
+
+        Assert.NotNull(output);
+        Assert.Equal(output, $"Orchestration version: {version}");
+    }
+
+    [Fact]
+    public async Task OrchestrationVersioning_MatchTypeNone_NoVersionFailure()
+    {
+        var workerVersion = "0.1";
+        var clientVersion = "0.2";
+        await using HostTestLifetime server = await this.StartWorkerAsync(b =>
+        {
+            b.AddTasks(tasks => tasks
+                .AddOrchestratorFunc<string, string>("Versioned_Orchestration", (ctx, input) =>
+                {
+                    return ctx.CallActivityAsync<string>("Versioned_Activity", ctx.Version);
```

Comment:
Pro-tip: you don't need to call an activity - you can just do the following:

```suggestion
                    return Task.FromResult(ctx.Version);
```

...not that there's anything wrong with what you're doing, but I thought I'd mention it if you wanted to remove a few lines of code from these tests.

Reply:
Good to know! I'll leave them as is for now as I don't mind some verbosity in a test.